src/System.IO.Compression.ZipFile/tests/ZipFileInvalidFileTests.cs

```diff
@@ -46,55 +46,85 @@ public void InvalidConstructors()
         [Fact]
         public void InvalidFiles()
         {
-            ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(bad("EOCDmissing.zip")));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.Open(bad("EOCDmissing.zip"), ZipArchiveMode.Update));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(bad("CDoffsetOutOfBounds.zip")));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.Open(bad("CDoffsetOutOfBounds.zip"), ZipArchiveMode.Update));
+            using (TempFile testArchive = CreateTempCopyFile(bad("EOCDmissing.zip"), GetTestFilePath()))
+            {
+                ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(testArchive.Path));
```

Comment:
> The only thing which makes we wonder is, what this Stream is created for

I now know why it is created. If the stream is not seekable the stream param get set to a new Stream which changes the pointer address. So only if this new Stream get created it should be disposed. It's just a check if a new Stream got created.